pkg/client/process_volumes_darwin.go

```diff
@@ -0,0 +1,51 @@
+package client
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/docker/cli/cli/compose/loader"
+	"github.com/docker/cli/cli/compose/types"
+	"github.com/pkg/errors"
+
+	"github.com/buildpacks/pack/internal/style"
+)
+
+func processVolumes(imgOS string, volumes []string) (processed []string, warnings []string, err error) {
+	for _, v := range volumes {
+		volume, err := parseVolume(v)
+		if err != nil {
+			return nil, nil, err
+		}
+		sensitiveDirs := []string{"/cnb", "/layers"}
+		if imgOS == "windows" {
+			sensitiveDirs = []string{`c:/cnb`, `c:\cnb`, `c:/layers`, `c:\layers`}
+		}
+		for _, p := range sensitiveDirs {
+			if strings.HasPrefix(strings.ToLower(volume.Target), p) {
+				warnings = append(warnings, fmt.Sprintf("Mounting to a sensitive directory %s", style.Symbol(volume.Target)))
+			}
+		}
+		mode := "ro"
+		if strings.HasSuffix(v, ":rw") && !volume.ReadOnly {
+			mode = "rw"
+		}
+		processed = append(processed, fmt.Sprintf("%s:%s:%s", volume.Source, volume.Target, mode))
+	}
+	return processed, warnings, nil
+}
+
+func parseVolume(volume string) (types.ServiceVolumeConfig, error) {
+	// volume format: '<host path>:<target path>[:<options>]'
+	split := strings.Split(volume, ":")
+	if len(split) == 3 {
+		if split[2] != "ro" && split[2] != "rw" && !strings.Contains(split[2], "volume-opt") {
+			return types.ServiceVolumeConfig{}, errors.New(fmt.Sprintf("platform volume %q has invalid format: invalid mode: %s", volume, split[2]))
+		}
+	}
```

Comment:
Just curious, what happens if we don't check this? It seems that `loader.ParseVolume` handles at least some of this logic (https://github.com/docker/cli/blob/0bdc20ecbeb510083b98344be99c2e4f66b2ec9f/cli/compose/loader/volume.go#L16)

Reply:
If I remove the validation, this [test](https://github.com/buildpacks/pack/pull/2136/files#diff-03220a2f3a516ece374582fcc687eb7c47a3c6e2dcd11e0f5a117c5a9c0decc6R2753) fails. 

The internal parsing method [does not](https://github.com/docker/cli/blob/0bdc20ecbeb510083b98344be99c2e4f66b2ec9f/cli/compose/loader/volume.go#L82) throw an error with an invalid format volume option

Reply:
Ah, makes sense. Okay, seems fine to me